src/material/form-field/form-field.ts

```diff
@@ -672,6 +695,23 @@ export class MatFormField
     )`;
   }
 
+  /**
+   * Updates the horizontal offset of the label in the outline appearance when the prefix containers are resized.
+   */
+  private _updateOutlineLabelOffsetOnPrefixResize() {
+    const iconPrefixContainer = this._iconPrefixContainer?.nativeElement;
+    const textPrefixContainer = this._textPrefixContainer?.nativeElement;
+
+    merge(
+      iconPrefixContainer ? this._resizeObserver.observe(iconPrefixContainer) : EMPTY,
+      textPrefixContainer ? this._resizeObserver.observe(textPrefixContainer) : EMPTY,
+    )
+      .pipe(takeUntil(this._prefixChildren.changes))
```

Comment:
I think we should save `this._prefixChildren.changes.pipe(map(prefixes => !!prefixes.length))` to a private field (e.g. `this._hasPrefix`), that way we can use it in the `takeUntil` here, because we don't want to stop when the count goes from 2 => 1, only when it goes to 0. Alternatively could just repeat the pipe logic here instead of saving it on a field.